### ISSUE TITLE
Report allocations

### ISSUE THREAD

**Issue by jtrakk** [aviatesk/JET.jl#286]
Julia docs say

> One of the most common techniques to improve performance is to reduce memory allocation.

Julia has some [memory-allocation analysis](https://docs.julialang.org/en/v1/manual/profile/#Memory-allocation-analysis) but it would be useful to have in JET.

**Comment by timholy** [comment]
JET is a static analyzer. There's no reasonable way that it could predict the amount of memory required, but it might be able to address whether memory would be allocated at all.

ProfileView makes it pretty easy to find "serious" allocation: just look for yellow bars.

**Comment by jtrakk** [comment]
> it might be able to address whether memory would be allocated at all.

Right. In some places I may want to ensure there will be no allocation. Highlighting those locations, just like `@report_opt`, would be helpful for this.

**Comment by jtrakk** [comment]
The relevant Julia issue is "A macro that forbids allocation – turning allocation into error" https://github.com/JuliaLang/julia/issues/34248

**Comment by kbarros** [comment]
See the related package: https://github.com/JuliaLang/AllocCheck.jl

**Comment by aviatesk** [comment]
Most of the time, reporting allocations using JET's abstract interpretation based approach probably won't be very helpful. I recommend using AllocCheck instead.
